qir/qat/Commandline/ConfigurationManager.hpp

```diff
@@ -21,6 +21,53 @@ namespace microsoft
 namespace quantum
 {
 
+    class ConfigurationManager;
+
+    class DeferredValue
+    {
+      public:
+        using DeferredValuePtr = std::shared_ptr<DeferredValue>;
+        using IConfigBindPtr   = std::shared_ptr<IConfigBind>;
+
+        static DeferredValuePtr create()
+        {
+            DeferredValuePtr ret;
+            ret.reset(new DeferredValue());
+            return ret;
+        }
+
+        void setReference(IConfigBindPtr const& value)
+        {
+            value_ref_ = value;
+        }
+
+        bool isDereferencable() const
+        {
+            return value_ref_ != nullptr;
+        }
```

Comment:
Nit.
```suggestion
        bool isDereferenceable() const
        {
            return value_ref_ != nullptr;
        }
```
[Details](https://en.wiktionary.org/wiki/referenceable).